### PR TITLE
SqlCredential - Separating port from host

### PR DESCRIPTION
**Pull request by eddynaka**

### CODE REVIEW

File: Src/Plugins/Tests.Security/TestData/SecurePlaintextSecrets/ExpectedOutputs/SEC101_037.SqlCredentials.sarif [hunk 1]
```diff
@@ -205,11 +205,11 @@
...
-            "ValidationFingerprintHash/v1": "d79c54c338cf6b8786b93457bee7bed8e7090750f873a0ce06ed4dca3431acd0",
-            "AssetFingerprint/v2": "{\"host\":\"servername2.database.windows.net,1433\",\"id\":\"username2\",\"part\":\"servers\",\"platform\":\"Azure\",\"resource\":\"database2\"}",
-            "ValidationFingerprint/v2": "{\"host\":\"servername2.database.windows.net,1433\",\"id\":\"username2\",\"resource\":\"database2\",\"secret\":\"password2\"}"
+            "AssetFingerprint/v1": "[host=servername2.database.windows.net][id=username2][part=servers][platform=Azure][port=1433][resource=database2]",
+            "ValidationFingerprint/v1": "[host=servername2.database.windows.net][id=username2][port=1433][resource=database2][secret=password2]",
+            "ValidationFingerprintHash/v1": "d8f735ee725f3c05e20b44ba9a8004f07330d1feb965d2ec56ee35f82edb22f6",
+            "AssetFingerprint/v2": "{\"host\":\"servername2.database.windows.net\",\"id\":\"username2\",\"part\":\"servers\",\"platform\":\"Azure\",\"port\":\"1433\",\"resource\":\"database2\"}",
```
Comment (michaelcfanning) [inline Src/Plugins/Tests.Security/TestData/SecurePlaintextSecrets/ExpectedOutputs/SEC101_037.SqlCredentials.sarif:211]: This is wrong, the asset fingerprint should not contain the port. The same asset accessed via two distinct ports, is a single asset.

File: Src/Plugins/Tests.Security/TestData/SecurePlaintextSecrets/ExpectedOutputs/SEC101_037.SqlCredentials.sarif [hunk 2]
```diff
@@ -835,11 +835,11 @@
             }
           ],
           "fingerprints": {
-            "AssetFingerprint/v1": "[host=servername3.database.windows.net,1433][id=username3][part=servers][platform=Azure][resource=databse3]",
-            "ValidationFingerprint/v1": "[host=servername3.database.windows.net,1433][id=username3][resource=databse3][secret=password3]",
-            "ValidationFingerprintHash/v1": "eeccc815a478632178473bd1bc7a2f96c633106af83c25ef7e52fa6a0de16c84",
-            "AssetFingerprint/v2": "{\"host\":\"servername3.database.windows.net,1433\",\"id\":\"username3\",\"part\":\"servers\",\"platform\":\"Azure\",\"resource\":\"databse3\"}",
-            "ValidationFingerprint/v2": "{\"host\":\"servername3.database.windows.net,1433\",\"id\":\"username3\",\"resource\":\"databse3\",\"secret\":\"password3\"}"
+            "AssetFingerprint/v1": "[host=servername3.database.windows.net][id=username3][part=servers][platform=Azure][port=1433][resource=databse3]",
```
Comment (michaelcfanning) [inline Src/Plugins/Tests.Security/TestData/SecurePlaintextSecrets/ExpectedOutputs/SEC101_037.SqlCredentials.sarif:838]: >databse3 [](http://example.com/codeflow?start=140&length=8)

can yo uchange this to 'database3'?

**Review by michaelcfanning — APPROVED** [review]
:shipit:

File: Src/Plugins/Tests.Security/TestData/SecurePlaintextSecrets/ExpectedOutputs/SEC101_037.SqlCredentials.sarif (hunk 1 above)
Comment (eddynaka) [reply Src/Plugins/Tests.Security/TestData/SecurePlaintextSecrets/ExpectedOutputs/SEC101_037.SqlCredentials.sarif]: Interesting! We weren't removing port from the AssetFingerprint.
Just made the change to enable this.

File: Src/Plugins/Tests.Security/TestData/SecurePlaintextSecrets/ExpectedOutputs/SEC101_037.SqlCredentials.sarif (hunk 2 above)
Comment (eddynaka) [reply Src/Plugins/Tests.Security/TestData/SecurePlaintextSecrets/ExpectedOutputs/SEC101_037.SqlCredentials.sarif]: Fixed!
Nice eyes!

**Pull request merged**